semcore/date-picker/src/components/DateRangeComparatorAbstract.jsx

```diff
@@ -0,0 +1,442 @@
+            this.setState({
+              dirtyValue: undefined,
+              dirtyCompare: undefined,
+              dirtyToggler: undefined,
+              range: 'value',
+            });
+            this.handlers.displayedPeriod(
+              getLatestDate(this.asProps.value ?? []) || this.props.defaultDisplayedPeriod,
+            );
+          }
+        },
+      ],
+      highlighted: null,
+      compareHighlighted: null,
+      value: [
+        null,
+        (value) => {
+          if (value?.[0] && value?.[1]) this.handlers.displayedPeriod(getLatestDate(value));
+        },
+      ],
+      compare: [
+        null,
+        (value) => {
+          if (value?.[0] && value?.[1]) this.handlers.displayedPeriod(getLatestDate(value));
+        },
+      ],
+    };
+  }
+
+  getApplyProps() {
+    const { value, compare, getI18nText } = this.asProps;
+    const { dirtyValue, dirtyCompare } = this.state;
+    return {
+      getI18nText,
+      onClick: () => this.handleApply(dirtyValue ?? value, dirtyCompare ?? compare),
+    };
+  }
+
+  getResetProps() {
+    const { getI18nText } = this.asProps;
+    return {
+      getI18nText,
+      onClick: () => this.handleApply(null, null),
+    };
+  }
+
+  handleApply = (value, compare) => {
+    this.handlers.value(value);
+    this.handlers.compare(compare);
+    this.handlers.visible(false);
+  };
+
+  getDefaultPeriods() {
+    const { getI18nText } = this.asProps;
+    const today = new Date(new Date().setHours(0, 0, 0, 0));
+    return [
+      {
+        children: getI18nText('lastMonth'),
+        value: [
+          dayjs(today).subtract(1, 'month').startOf('month').toDate(),
+          dayjs(today).startOf('month').toDate(),
+        ],
+      },
+      {
+        children: getI18nText('last3Months'),
+        value: [
+          dayjs(today).subtract(2, 'month').startOf('month').toDate(),
+          dayjs(today).startOf('month').toDate(),
+        ],
+      },
+      {
+        children: getI18nText('last6Months'),
+        value: [
+          dayjs(today).subtract(5, 'month').startOf('month').toDate(),
+          dayjs(today).startOf('month').toDate(),
+        ],
+      },
+      {
+        children: getI18nText('last12Months'),
+        value: [
+          dayjs(today).subtract(11, 'month').startOf('month').toDate(),
+          dayjs(today).startOf('month').toDate(),
+        ],
+      },
+    ];
+  }
+
+  getTitleProps(props, index) {
+    const { locale, displayedPeriod } = this.asProps;
+    return {
+      children: new Intl.DateTimeFormat(locale, { month: 'long', year: 'numeric' }).format(
+        dayjs(displayedPeriod).add(index, this.navigateStep).startOf(this.navigateStep).toDate(),
+      ),
+    };
+  }
+
+  getTriggerProps() {
+    const { value, compare, locale, separator = 'vs.' } = this.asProps;
+    const formattingProps = {
+      locale,
+      day: 'numeric',
+      month: 'short',
+      year: 'numeric',
+    };
+    let children = 'Select date ranges';
```

Comment:
i18n?

Reply:
i18n here is not required but yes, it should be different and come from placeholder prop